### PR TITLE
rabbitmq warning reports connection_closed_abruptly due to celery ping heartbeat

### PR DESCRIPTION
**Pull request by jeroenmaelbrancke**
[issue 558](https://github.com/openvstorage/framework/issues/558)

### CODE REVIEW

File: ovs/extensions/generic/heartbeat.py [hunk 1]
```diff
@@ -50,21 +50,23 @@ def pulse():
...
+            if worker_states.get('celery@{0}'.format(node.name), {}).get('ok') == 'pong':
                 node.heartbeats['celery'] = current_time
             if node.machine_id == machine_id:
                 node.heartbeats['process'] = current_time
             else:
                 try:
                     # check timeout of other nodes and clear arp cache
                     if node.heartbeats and 'process' in node.heartbeats:
-                        if current_time - node.heartbeats['process'] >= HeartBeat.ARP_TIMEOUT:
+                        if current_time - node.heartbeats['process'] >= 30:
```
Comment (khenderick) [inline ovs/extensions/generic/heartbeat.py]: Why changing this to `30` instead of using the constant?

File: ovs/extensions/generic/heartbeat.py [hunk 2]
```diff
@@ -50,21 +50,23 @@ def pulse():
                                             EtcdConfiguration.get('/ovs/framework/messagequeue|password'),
                                             EtcdConfiguration.get('/ovs/framework/hosts/{0}/ip'.format(machine_id)))
 
-        celery_path = OSManager.get_path('celery')
-        worker_states = check_output("{0} inspect ping -b {1} --timeout=5 2> /dev/null | grep OK | perl -pe 's/\x1b\[[0-9;]*m//g' || true".format(celery_path, amqp), shell=True)
+        celery = Celery(broker=amqp)
```
Comment (khenderick) [inline ovs/extensions/generic/heartbeat.py]: Maybe using a `with` statement, so the close is always executed.

File: ovs/extensions/generic/heartbeat.py [hunk 3]
```diff
@@ -50,21 +50,23 @@ def pulse():
...
+        worker_states = celery.control.inspect().ping()
```
Comment (khenderick) [inline ovs/extensions/generic/heartbeat.py]: Are we sure this call finished within a reasonable time (especially if the RMQ cluster is borked up)?

Comment (jeroenmaelbrancke) [inline ovs/extensions/generic/heartbeat.py]: Default timeout is 1 sec. [celery.app.control](http://docs.celeryproject.org/en/latest/reference/celery.app.control.html)

**Pull request merged**